api/envoy/config/filter/listener/original_src/v2alpha1/original_src.proto

```diff
@@ -0,0 +1,28 @@
+syntax = "proto3";
+
+package envoy.config.filter.listener.original_src.v2alpha1;
+option java_package = "io.envoyproxy.envoy.config.filter.listener.original_src.v2alpha1";
+option java_multiple_files = true;
+
+option go_package = "v2alpha1";
+
+import "validate/validate.proto";
+
+// [#protodoc-title: Original Src Filter]
+// Use the Original source address on upstream connections.
+// TODO(klarose): Add ref to docs
```

Comment:
Do you plan on doing the full doc flow in a follow up? We should have complete config docs for this filter and it would be great to have some more general overview of all of the features we have that help with transparency?

Reply:
That said, I haven't covered the last bit -- an architectural overview of the feature(s). I can certainly do that as part of that PR.

Reply:
Sounds good on a follow up. Yes please on some type of arch overview!